scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -773,30 +773,49 @@ def test_input_types(self):
     @pytest.mark.slow
     @pytest.mark.timeout(120)  # prerelease_deps_coverage_64bit_blas job
     @sup_sparse_efficiency
-    def test_random(self):
-        def random_triangle_matrix(n, lower=True):
-            A = scipy.sparse.random(n, n, density=0.1, format='coo')
+    @pytest.mark.parametrize("n", [10, 10**2, 10**3])
+    @pytest.mark.parametrize("m", [1, 10])
+    @pytest.mark.parametrize("lower", [True, False])
+    @pytest.mark.parametrize("format", ["csr", "csc"])
+    @pytest.mark.parametrize("unit_diagonal", [False, True])
+    @pytest.mark.parametrize("choice_of_A", ["real", "complex"])
+    @pytest.mark.parametrize("choice_of_b", ["floats", "ints", "complexints"])
+    def test_random(self, n, m, lower, format, unit_diagonal, choice_of_A, choice_of_b):
+        def random_triangle_matrix(n, lower=True, format="csr", choice_of_A="real"):
+            if choice_of_A == "real":
+                dtype = np.float64
+            elif choice_of_A == "complex":
+                dtype = np.complex128
+            else:
+                raise ValueError("choice_of_A must be 'real' or 'complex'.")
+            rng = np.random.default_rng(789002319)
+            rvs = rng.random
+            A = scipy.sparse.random(n, n, density=0.1, format='lil', dtype=dtype,
+                    random_state=rng, data_rvs=rvs)
             if lower:
-                A = scipy.sparse.tril(A)
+                A = scipy.sparse.tril(A, format="lil")
             else:
-                A = scipy.sparse.triu(A)
-            A = A.tocsr(copy=False)
+                A = scipy.sparse.triu(A, format="lil")
             for i in range(n):
                 A[i, i] = np.random.rand() + 1
+            if format=="csc":
```

Comment:
```suggestion
            if format == "csc":
```

Sorry about all the linting issues, scipy is unfortunately not consistent there at the moment.

Reply:
Sorry for missing this, I fixed it in a1020556646a0259dc2780bf564c5947f2d24f41 as well.

Reply:
No worries, happens to all of us.